packages/drivers/odsp-driver/src/vroom.ts

```diff
@@ -47,14 +47,20 @@ export async function fetchJoinSession(
 ): Promise<ISocketStorageDiscovery> {
     const token = await getStorageToken(options, "JoinSession");
 
-    const extraProps = options.refresh
-        ? { hasClaims: !!options.claims, hasTenantId: !!options.tenantId }
-        : {};
+    const details: ITelemetryProperties = {
+        refreshedToken: options.refresh,
+        requestSocketToken,
+    };
+    if (options.refresh) {
+        details.hasClaims = !!options.claims;
+        details.hasTenantId = !!options.tenantId;
+    }
+
     return PerformanceEvent.timedExecAsync(
         logger, {
             eventName: "JoinSession",
             attempts: options.refresh ? 2 : 1,
-            ...extraProps,
+            details: JSON.stringify(details),
```

Comment:
This would be harder to parse (in telemetry), and pretty much everywhere we expand properties (though in some cases we add prefix to such properties to avoid conflicts / make it easier to understand where it came from)

Reply:
I'm open to reverting this, but let me explain first. We have SO MANY columns in our logs. Some have lots of overlap but there is a long tail of columns that are entirely specific to a particular event.

When browsing a session log, the common columns are of great value, you can glean a lot from selecting them and looking through. The event-specific ones are lost. So I've started using this details property in some PRs and putting JSON in it so we can include this in common queries for browsing a session's logs. 

Where this makes more work is when you are analyzing a specific event, with a priori knowledge of exactly what props it has and what they mean. But this is not hard, Kusto has excellent support for parsing JSON out into new columns.

What do you think?

Reply:
I added back the other properties, in addition to the details JSON, to try it out.

Reply:
Sounds good. I like less columns :) But we should be consistent, so maybe worth sending email and looking at biggest offenders to move to this new model?

Reply:
Yeah something I will follow up on over time, and share learnings/guidance.